### ISSUE TITLE
Getting FactoryBot not registered for some factories while others are working

### ISSUE THREAD

**Issue by aditya01933** [thoughtbot/factory_bot_rails#268]
In my factories, this one is working:

spec/factories/cities.rb
```
FactoryBot.define do
  factory :city do
    name { Faker::Address.city }
    state { create(:state) }
  end
end
```

while this one is not working:
/spec/factories/admins.rb
``` 
FactoryBot.define do
  factory :admin do
    sequence(:email) { |n| "email-#{n}@example.com" }
    phone { rand(10 ** 10) }
    password '12345678'
    avatar { create(:image) }
    first_name "first_name"
    last_name "last_name"
  end
end
```

I am not able to understand why this could happen, please help me solve/debug the problem.

I am on the latest version of the gem.

These factories are working when I am explicitly requiring them.

**Comment by mfbmina** [comment]
Which test framework are you using? Can you give more information such as spec_helper?

**Comment by aditya01933** [comment]
Yes, I am using rspec and it's working when I am requiring the files in spec_helper

**Comment by taufnrsyd** [comment]
+1

Experiencing the exac same problem, I'm also using RSpec. My `User` factory works,

```
# spec/factories/user.rb

FactoryBot.define do
  factory :user do
    email { Faker::Internet.email }
    password 'password'
    password_confirmation 'password'
  end
end
```

while my `Link` factory does not.

```
# spec/factories/link.rb

FactoryBot.define do
  factory :link do
    address Faker::Bitcoin::address
    link Faker::Internet.slug
  end
end
```

Tried with `create :link` as well as `create_list :link`, none works. Below is a gist of my RSpec config, with part related to my test configuration.

https://gist.github.com/taufnrsyd/0f56478dc8463103694f8115e23e20a9

**Comment by taufnrsyd** [comment]
+1

Experiencing the exact same problem, I'm also using RSpec. My `User` factory works,

```
# spec/factories/user.rb

FactoryBot.define do
  factory :user do
    email { Faker::Internet.email }
    password 'password'
    password_confirmation 'password'
  end
end
```

while my `Link` factory does not.

```
# spec/factories/link.rb

FactoryBot.define do
  factory :link do
    address Faker::ParksAndRec.city
    link Faker::Internet.slug
  end
end
```

Tried with `create :link` as well as `create_list :link`, none works. Below is a gist of my RSpec config, with part related to my test configuration.

https://gist.github.com/taufnrsyd/0f56478dc8463103694f8115e23e20a9

**Comment by victorhazbun** [comment]
This should be documented in the project Readme.

> Found the problem!! In my spec_helper. I was requiring `factory_bot`, changing it to require `factory_bot_rails` worked for me.

**Comment by aoyshi** [comment]
Was stuck on the same issue for a while for `rails 6.1.7.3` and `factory_bot_rails 6.2.0`. Turns out, I had the `/factories` folder in the wrong location, relative to the paths that FactoryBot was searching for loading. I didn't need the explicit reload in `rails_helper.rb` or `spec_helper.rb`

If you spin up `rails c` for your project and run `FactoryBot.reload`, you'll see the array of file paths that FactoryBot is looking for to find your factories. Mine was nested one level above all the paths, and I just had to move it so it complied with one of them and still made sense for my project structure.
